### PR TITLE
Do not merge - develop branch review

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 1]
```diff
@@ -88,23 +89,19 @@ private fun AppDrawer(
     currentScreen: Screen,
     closeDrawer: () -> Unit
 ) {
-    Column(
-        crossAxisSize = LayoutSize.Expand,
-        mainAxisSize = LayoutSize.Expand
-    ) {
+    Column(modifier = ExpandedHeight wraps ExpandedWidth) {
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: I think there should be an "Expanded" that does both. (not 100% sure)

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 2]
```diff
@@ -126,45 +123,54 @@ private fun AppDrawer(
...
 private fun DrawerButton(
+    modifier: Modifier = Modifier.None,
     @DrawableRes icon: Int,
     label: String,
     isSelected: Boolean,
     action: () -> Unit
 ) {
     val textIconColor = if (isSelected) {
-        +themeColor { primary }
+        (+MaterialTheme.colors()).primary
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: Extract to a val and reuse? something like
`val colors = (+MaterialTheme.colors())`

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 3]
```diff
@@ -126,45 +123,54 @@ private fun AppDrawer(
...
+                )
             }
         }
     }
 }
+
+@Preview
+@Composable
+fun preview() {
+    //DrawerButton(icon = R.drawable.ic_home, label= "Preview", isSelected = true, action = {})
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: Remove comment

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt [hunk 4]
```diff
@@ -130,10 +119,8 @@ private fun BottomBarAction(
         radius = 24.dp
     ) {
         Clickable(onClick = onClick) {
-            Padding(12.dp) {
-                Container(width = 24.dp, height = 24.dp) {
-                    DrawVector(+vectorResource(id))
-                }
+            Container(modifier = Spacing(12.dp), width = 24.dp, height = 24.dp) {
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt]: Can also width and height be modifiers?

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt [hunk 5]
```diff
@@ -203,29 +202,29 @@ private data class ParagraphStyling(
 )
 
 private fun ParagraphType.getTextAndParagraphStyle(): ParagraphStyling {
-    var textStyle: TextStyle = +themeTextStyle { body1 }
+    var textStyle: TextStyle = (+MaterialTheme.typography()).body1
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: extract `(+MaterialTheme.typography())` to a val. Same for other places in the PR

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt [hunk 6]
```diff
@@ -46,29 +46,26 @@ fun PostCardPopular(post: Post) {
...
+                    Column(modifier = Expanded) {
                         val image = post.image ?: +imageResource(R.drawable.placeholder_4_3)
                         Container(height = 100.dp, expanded = true) {
                             DrawImage(image)
                         }
                         Column(modifier = Spacing(16.dp)) {
                             Text(
                                 text = post.title,
-                                style = (+themeTextStyle { h6 }).withOpacity(0.87f),
+                                style = ((+MaterialTheme.typography()).h6).withOpacity(0.87f),
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt]: Same as before, extract to a val

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt [hunk 7]
```diff
@@ -168,13 +167,12 @@ private fun CodeBlockParagraph(
...
     ) {
-        Padding(16.dp) {
-            Text(
-                text = text,
-                style = textStyle,
-                paragraphStyle = paragraphStyle
-            )
-        }
+        Text(
+            modifier = Spacing(top = 16.dp),
```
Comment (acbellini) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: Are you sure about spacing top here? The padding was all around the text

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 8]
```diff
@@ -131,14 +130,14 @@ private fun DrawerButton(
     action: () -> Unit
 ) {
     val textIconColor = if (isSelected) {
-        +themeColor { primary }
+        (+MaterialTheme.colors()).primary
```
Comment (Splogy) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: Can all these +MaterialTheme.colors() in this block be extracted into one? I.e:

```
val colors = +MaterialTheme.colors()

if () {
    colors.primary
} else {
    colors.onSurface.copy..
}
```

etc

Same throughout these changes and for typography too

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardTop.kt [hunk 9]
```diff
@@ -48,7 +48,7 @@ import com.example.jetnews.ui.themeTypography
 fun PostCardTop(post: Post) {
     // TUTORIAL CONTENT STARTS HERE
     Padding(16.dp) {
-        Column(crossAxisSize = LayoutSize.Expand) {
+        FlexColumn(crossAxisSize = LayoutSize.Expand) {
```
Comment (Splogy) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardTop.kt]: Should still be a `Column`, `FlexColumn` and `Column` have been merged into one, using modifiers to define flexible behavior. (default is inflexible). Ditto elsewhere / for any `FlexRows`.

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt [hunk 10]
```diff
@@ -56,18 +57,18 @@ fun PostCardPopular(post: Post) {
...
```
Comment (Splogy) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt]: Ditto with extracting typography above, no need to access the ambient value every time

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 11]
```diff
@@ -87,22 +89,19 @@ private fun AppDrawer(
     currentScreen: Screen,
     closeDrawer: () -> Unit
 ) {
-    FlexColumn(
-        crossAxisSize = LayoutSize.Expand
-    ) {
+    Column(modifier = ExpandedHeight wraps ExpandedWidth) {
```
Comment (nastia05) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: better to replace with one modifier instead 
Column(modifier = Expanded)

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 12]
```diff
@@ -87,22 +89,19 @@ private fun AppDrawer(
...
+            VectorImage(
+                id = R.drawable.ic_jetnews_logo,
+                tint = (+MaterialTheme.colors()).primary
+            )
+            WidthSpacer(8.dp)
+            VectorImage(id = R.drawable.ic_jetnews_wordmark)
         }
         Divider(color = Color(0x14333333))
         DrawerButton(
+            modifier = Inflexible ,
```
Comment (nastia05) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]: it's Inflexible by default. You don't specify it in other children so should be omitted here as well

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt (hunk 7 above)
Comment (JoseAlcerreca) [reply JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: Good catch

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt [hunk 13]
```diff
@@ -157,33 +152,26 @@ private fun TabWithSections(
...
-            inflexible {
-                val selected = isTopicSelected(topicKey)
-                SelectTopicButton(
-                    onSelected = {
-                        selectTopic(topicKey, !selected)
-                    },
-                    selected = selected
-                )
+    Row(
+        Spacing(left = 16.dp, right = 16.dp)
```
Comment (nastia05) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt]: named params are used elsewhere so maybe better to keep consistency here as well?
Row(modifier = Spacing...

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt [hunk 14]
```diff
@@ -141,13 +143,12 @@ private fun Paragraph(paragraph: Paragraph) {
...
             ParagraphType.Header -> {
-                Padding(top = 16.dp) {
-                    Text(
-                        text = annotatedString,
-                        style = textStyle,
-                        paragraphStyle = paragraphStyle
-                    )
-                }
+                Text(
+                    modifier = Spacing(16.dp),
```
Comment (nastia05) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: Spacing(top = 16.dp) :)

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 15]
```diff
@@ -106,29 +94,20 @@ fun PostCardHistory(post: Post) {
...
-                            PostTitle(post = post)
-                            AuthorAndReadTime(post)
-                        }
-                    }
-                    inflexible {
-                        Padding(top = 8.dp, bottom = 8.dp) {
-                            VectorImage(R.drawable.ic_more)
-                        }
-                    }
+            Row(Spacing(all = 16.dp)) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]:
```suggestion
            Row(modifier = Spacing(all = 16.dp)) {
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt [hunk 16]
```diff
@@ -184,15 +184,16 @@ private fun BulletParagraph(
...
-    FlexRow {
-        inflexible {
-            Container(width = 8.dp, height = 8.dp) {
-                DrawShape(shape = CircleShape, color = Color.DarkGray)
-            }
-        }
-        flexible(flex = 1f) {
-            Text(text = text, style = textStyle, paragraphStyle = paragraphStyle)
+    Row {
+        Container(width = 8.dp, height = 8.dp) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: Should these be modifiers as well?

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt [hunk 17]
```diff
@@ -59,25 +58,21 @@ fun InterestsScreen(openDrawer: () -> Unit) {
...
                 }
             }
+        )
+        TabRow(items = sectionTitles, selectedIndex = section.ordinal) { index, text ->
+            Tab(text = text, selected = section.ordinal == index) {
+                section = Sections.values()[index]
+            }
         }
-        expanded(1f) {
+        Container(Flexible(1f)) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt]:
```suggestion
        Container(modifier = Flexible(1f)) {
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 18]
```diff
@@ -40,12 +40,13 @@ import com.example.jetnews.ui.lightThemeColors
...
 fun SelectTopicButton(
+    modifier: Modifier = Modifier.None,
     onSelected: ((Boolean) -> Unit)? = null,
     selected: Boolean = false
 ) {
     Ripple(bounded = false) {
-        Toggleable(checked = selected, onCheckedChange = onSelected) {
-            Container(width = 36.dp, height = 36.dp) {
+        Toggleable(selected, onSelected) {
+            Container(modifier = modifier, width = 36.dp, height = 36.dp) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt]: Again, should the Width/Height be modifiers?

File: JetNews/build.gradle [hunk 19]
```diff
@@ -33,5 +33,6 @@ allprojects {
         google()
         jcenter()
         maven { url 'https://dl.bintray.com/kotlin/kotlin-eap' }
+        maven { url 'https://ci.android.com/builds/submitted/6029545/androidx_snapshot/latest/ui/repository/' }
```
Comment (nickbutcher) [inline JetNews/build.gradle]:
```suggestion
        maven { url 'https://ci.android.com/builds/submitted/6033284/androidx_snapshot/latest/ui/repository/' }
```

**Pull request merged**